src/ddmd/func.d

```diff
@@ -156,6 +156,61 @@ enum FUNCFLAGinlineScanned    = 0x20;   /// function has been scanned for inline
 enum FUNCFLAGinferScope       = 0x40;   /// infer 'scope' for parameters
 
 
+/****************************************************
+ * Determine whether an 'in' contract is declared inside
+ * the given function or any of its overrides.
+ * Params:
+ *      fd = the function to search
+ * Returns:
+ *      true    found an 'in' contract
+ *      false   didn't find one
+ */
+private bool needsFrequire(FuncDeclaration fd)
+{
+    if (fd.frequire)
+        return true;
+
+    for (size_t i = 0; i < fd.foverrides.dim; i++)
+    {
+        FuncDeclaration fdv = fd.foverrides[i];
+
+        if (fdv.frequire)
+            return true;
```

Comment:
These two lines are not strictly necessary, as 170+171 will test it.

Reply:
This is called prior to merging `fensure`, so I thought that it would be needed.